lib/hidDevice.js

```diff
@@ -81,15 +102,32 @@ module.exports = function(config) {
       arr.unshift(0); // The first byte of arr must contain the Report ID.
                       // As we only work with a single report, this is set to 0x00.
       connection.write(arr);
+      clearTimeout(abortTimer);
       callback();
     }
   }
 
+  function sendPromisified(bytes) {
+    return new Promise((resolve, reject) => {
+      send(bytes, function(err) {
+        if (err) {
+          // exceptions inside promise aren't automatically thrown,
+          // so we have to reject them here
+          reject(err);
+        }
+        resolve();
```

Comment:
Should this be in an `else` clause? Is it otherwise simultaneously calling `reject` and `resolve`?